server/routes/api/messages.js

```diff
@@ -43,4 +51,88 @@ router.post("/", async (req, res, next) => {
   }
 });
 
+// Update mssage read status in conversation
+router.patch("/", async (req, res, next) => {
+  try {
+    if (!req.user) {
+      return res.sendStatus(401);
+    }
+
+    const { senderId, activeConversationId } = req.body;
+
+    // check if activeConversationId is exist
+    if (activeConversationId) {
+      let conversation = await Conversation.findConversationById(
+        activeConversationId,
+        senderId,
+        req.user.id
+      );
+
+      // if not exist return error
+      if (!conversation) return res.json(conversation);
+
+      // if activeConversationId exists, start update messages
+      const messages = await Message.findAll({
+        where: {
+          conversationId: activeConversationId,
+          senderId: {
+            [Op.eq]: senderId,
+          },
+          readStatus: {
+            [Op.is]: false,
+          },
+        },
+      });
+
+      // there is no messages, then return empty array
+      if (!Array.isArray(messages) || !messages.length)
+        return res.json({ messages });
+
+      // if have messages, update read message status
+      messages.forEach((message) => {
+        if (!message.readStatus) message.readStatus = true;
+        message.readById = senderId;
+        message.readTime = Date.now();
+      });
+
+      // save updated messages
+      await Promise.all(messages.map(async (message) => await message.save()));
+      return res.json({ messages });
+    }
+
+    let conversation = await Conversation.findConversation(
+      senderId,
+      req.user.id
+    );
+
+    // if not found conversation return error
+    if (!conversation) return res.json(conversation);
+
+    const messages = await Message.findAll({
+      where: {
+        conversationId: activeConversationId,
+        senderId: {
+          [Op.ne]: senderId,
+        },
+      },
+    });
+
+    // if array messages is empty
+    if (!Array.isArray(messages) || !messages.length)
+      return res.json({ messages });
+
+    // update massage status
+    messages.forEach((message) => {
+      if (!message.readStatus) message.readStatus = true;
+      message.readById = senderId;
+      message.readTime = Date.now();
+    });
+
+    await Promise.all(messages.map(async (message) => await message.save()));
+    return res.json({ messages });
```

Comment:
You shouldn't need to have all this duplicated logic for the scenario where a conversationId is not passed in - this endpoint should only work with a valid conversationId, so if it doesn't get passed in, you can return an error.